packages/mui-base/src/useOption/useOption.ts

```diff
@@ -44,16 +53,37 @@ export function useOption<Value>(params: UseOptionParameters<Value>): UseOptionR
 
   const { index } = useCompoundItem<Value, SelectOption<Value>>(value, selectOption);
 
-  const handleRef = useForkRef(optionRefParam, optionRef)!;
+  const handleRef = useForkRef(optionRefParam, optionRef, buttonRefHandler)!;
+
+  const createHandleKeyDown =
+    (otherHandlers: EventHandlers) => (event: React.KeyboardEvent & MuiCancellableEvent) => {
+      otherHandlers.onKeyDown?.(event);
+      if (event.defaultMuiPrevented) {
+        return;
+      }
+
+      if ([' ', 'Enter'].includes(event.key)) {
+        event.defaultMuiPrevented = true; // prevent listbox onKeyDown
```

Comment:
I'm wondering if it shouldn't be `preventDefault` to make sure selecting an option doesn't submit a form, for example.

Reply:
The `Enter` click is handled by `useButton`'s non-native code: https://github.com/mui/material-ui/blob/master/packages/mui-base/src/useButton/useButton.ts#L147

So `preventDefault` is called.